README.md

```diff
@@ -1,2 +1,119 @@
+user2 = Neo4j::Http::Node.new(label: "User", uuid: SecureRandom.uuid)
+relationship = Neo4j::Http::Relationship.new(label: "KNOWS")
+Neo4j::Http::Client.upsert_relationship(relationship: relationship, from: user1, to: user2, create_nodes: true)
+
+# Find an existing relationship
+Neo4j::Http::Client.find_relationship(relationship: relationship, from: user1, to: user2)
+
+# Delete the relationship if it exists
+Neo4j::Http::Client.delete_relationship(relationship: relationship, from: user1, to: user2)
+```
+
+Each of the methods exposed on `Neo4j::Http::Client` above are provided by instances of each of the following adapters:
+* `Neo4j::Http::CypherClient` - provides an `execute_cypher` method which sends raw cypher commands to neo4j
+* `Neo4j::Http::NodeClient` - provides a higher level API for upserting and deleting Nodes
+* `Neo4j::Http::RelationshipClient` - provides a higher levle API for upserting and deleting Relationships
+
+The Node and Relationship clients use the `CypherClient` under the hood.  Each of these provides simple access via a `default` class method, which uses the default `Neo4j::Http.config` for creating the connections. For example
+
+`Neo4j::Http::NodeClient.default`
+
+is equivalent to:
+
+```
+config = Neo4j::Http.config
+cypher_client = Neo4j::Http::CypherClient.new(config)
+node_client = Neo4j::Http::NodeClient.new(cypher_client)
+```
+
+to connect to a different Neo4j datbase, you can create a custom configuration like:
+```
+config = Neo4j::Http::Configuration.new({ datbase_name: 'test' })
+cypher_client = Neo4j::Http::CypherClient.new(config)
+node_client = Neo4j::Http::NodeClient.new(cypher_client)
+```
+
+## Versioning
+
+This project follows [semantic versioning](https://semver.org).
+
+## Development
+
+After checking out the repo, run `bin/setup` to install dependencies.
+
+To run specs, you'll need a running neo4j instance available at `localhst:7474`.  If you have Docker installed, this is easily done by using the provided [docker-file](https://github.com/doximity/neo4j-http/blob/master/docker-compose.yml) - simply run `docker-compose up` within the project directory, and once running, you can then, run `rake spec` to run the tests in another terminal.
+
+You can also run `bin/console` for an interactive prompt that will allow you to experiment.
+
+To install this gem onto your local machine, run `bundle exec rake install`.
+
+To release a new version, update the version number in `version.rb`, issue a pull request, and once merged with passing CI, the new gem version will be pushed automatically.
+
+## Contributing
+
+1. Fork it
+2. Create your feature branch (`git checkout -b my-new-feature`)
+3. Commit your changes (`git commit -am 'Add some feature'`)
+4. Push to the branch (`git push origin my-new-feature`)
+5. Create a new Pull Request
+6. Sign the CLA if you haven't yet. See [CONTRIBUTING.md](https://github.com/doximity/neo4j-http/blob/master/CONTRIBUTING.md)
+
```

Comment:
Can we add a section here about how to run any tests?

Reply:
That's included here: https://github.com/doximity/neo4j-http/pull/1/files/00e99e7340caf09e933c4a3924ce8f0ab2118848#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R100